test/configuration_unittest.cc

```diff
@@ -0,0 +1,81 @@
+#include "../src/configuration.h"
+#include "gtest/gtest.h"
+
+namespace google {
+
+class MetadataAgentConfigurationTest : public ::testing::Test {
+ public:
+  MetadataAgentConfiguration config;
+  static void ParseConfiguration(MetadataAgentConfiguration& config, const std::string configString) {
+    std::stringstream stream(configString);
+    config.ParseConfiguration(stream);
+  }
+};
+
+static void TestDefaultConfig(MetadataAgentConfiguration& config) {
```

Comment:
Just noticed (sorry): it's best to not start non-test methods with `Test`. How about `VerifyDefaultConfig` or `CheckDefaultConfig` (I have a slight preference for the former, but will be ok with either).